### PR TITLE
Update orthofinder and pick up new orthogroup output

### PR DESCRIPTION
**Pull request by TomHarrop**
Pick up new ["Hierarchical Orthogroups"](https://github.com/davidemms/OrthoFinder#phylogenetic-hierarchical-orthogroups-directory) output

Fixed  bioconda recipe (https://github.com/bioconda/bioconda-recipes/pull/45149) and update orthofinder to 2.5.5 (closes #5312)

---

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/orthofinder/orthofinder_only_groups.xml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-    </xrefs>
     <macros>
         <import>macros.xml</import>
     </macros>
+    <xrefs>
+        <xref type="bio.tools">OrthoFinder</xref>
+    </xrefs>
     <requirements>
         <requirement type="package" version="@TOOL_VERSION@">orthofinder</requirement>
-        <requirement type="package" version="2.36">util-linux</requirement>
```
Comment (bernt-matthias) [inline tools/orthofinder/orthofinder_only_groups.xml:11]: Tests should show if we can really remove this. Has been added here to allow for containerized tests: https://github.com/galaxyproject/tools-iuc/pull/2757

Comment (TomHarrop) [reply tools/orthofinder/orthofinder_only_groups.xml:11]: Oops. That was a mistake. But it passed the tests using this singularity container (and no other dependencies): https://github.com/bioconda/bioconda-recipes/pull/45149#issuecomment-1884242199

Thanks for spotting that, I will test it again once the new container is on quay.

Comment (TomHarrop) [reply tools/orthofinder/orthofinder_only_groups.xml:11]: @bernt-matthias this passes tests with the new biocontainer, run in Apptainer, without the `util-linux` requirement.

Do you think I should leave `util-linux` out?

**Review by bgruening — APPROVED** [review]
@TomHarrop is this still Draft?

**Comment by TomHarrop** [comment]
Yes. According to the manual the main output been deprecated and I have to pick up the new output in the wrapper.

**Comment by TomHarrop** [comment]
Thanks @bgruening, all done now

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**